bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/SseStatesResource.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Context
+    private HttpServletResponse response;
+
+    @Context
+    private HttpServletRequest request;
+
+    public SseStatesResource() {
+        this.executorService = Executors.newSingleThreadExecutor();
+        this.broadcaster = new ItemStateChangesSseBroadcaster();
+    }
+
+    /**
+     * Subscribes the connecting client for state updates. It will initially only send a "ready" event with an unique
+     * connectionId that the client can use to dynamically alter the list of tracked items.
+     *
+     * @return {@link EventOutput} object associated with the incoming
+     *         connection.
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @GET
+    @Produces(SseFeature.SERVER_SENT_EVENTS)
+    @ApiOperation(value = "Initiates a new item state tracker connection", response = EventOutput.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    public Object getStateEvents() throws IOException, InterruptedException {
+
+        final SseStateEventOutput eventOutput = new SseStateEventOutput();
+        broadcaster.add(eventOutput);
+
+        // Disables proxy buffering when using an nginx http server proxy for this response.
+        // This allows you to not disable proxy buffering in nginx and still have working sse
+        response.addHeader(X_ACCEL_BUFFERING_HEADER, "no");
+
+        // We want to make sure that the response is not compressed and buffered so that the client receives server sent
+        // events at the moment of sending them.
+        response.addHeader(HttpHeaders.CONTENT_ENCODING, "identity");
+
+        return eventOutput;
+    }
+
+    /**
+     * Alters the list of tracked items for a given state update connection
+     *
+     * @param connectionId the connection Id to change
+     * @param itemNames the list of items to track
+     */
+    @POST
+    @Path("{connectionId}")
+    @ApiOperation(value = "Changes the list of items a SSE connection will receive state updates to.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "Unknown connectionId") })
+    public Object updateTrackedItems(@PathParam("connectionId") String connectionId,
+            @ApiParam("items") Set<String> itemNames) {
+
+        try {
+            broadcaster.updateTrackedItems(connectionId, itemNames);
+        } catch (IllegalArgumentException e) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+
+        return Response.ok().build();
+    }
+
+    /**
+     * Broadcasts a state event to all currently listening clients, after transforming it to a simple map.
+     *
+     * @param stateChangeEvent the {@link ItemStateChangedEvent} containing the new state
+     */
+    public void broadcastEvent(final ItemStateChangedEvent stateChangeEvent) {
+        executorService.execute(new Runnable() {
```

Comment:
How about using `() -> {}` syntax instead of the Runnable?

Reply:
Same comment as above
https://github.com/openhab/openhab-core/blob/730a8e1e6104d966584b823819bb730f2695dde8/bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/SseResource.java#L126-L129